docs/src/manual/grid.md

```diff
@@ -3,8 +3,86 @@ DocTestSetup = :(using JuAFEM)
 ```
 
 # Grid
-TODO: Describe the grid format, and how to add sets etc.
 
-!!! note
+In JuAFEM a Grid is a collection of `Node`s and `Cell`s and is parameterized in its physical dimensionality and cell type.
+`Node`s are points in the physical space and can be initialized by a N-Tuple, where N corresponds to the dimensions.
 
-    Write some conversion functions from e.g. Abaqus file format.
+```julia
+n1 = Node((0.0, 0.0))
+```
+
+`Cell`s are defined based on the `Node` IDs. Hence, they collect IDs in a N-Tuple.
+Consider the following 2D mesh:
+
+![global mesh](./assets/global_mesh.svg)
+
+The cells of the grid can be described in the following way
+
+```julia
+julia> elements = [
+              (1,2,5,4),
+              (2,3,6,5),
+              (4,5,8,7),
+              (5,6,9,8)
+       ]
+```
+
+Additionally, the data structure `Grid` can hold node-, face- and cellsets. 
+All of these three sets are defined by a dictionary that maps a string key to a `Set`. 
+For the special case of node- and cellsets the dictionary's value is of type `Set{Int}`, i.e. a keyword is mapped to a node or cell ID, respectively. 
+
+
+Facesets are a more elaborate construction. They map a `String` key to a `Set{Int, Int}` consisting of `(global_cell_id, local_face_id)`. In order to understand the `local_face_id` properly, one has to consider the reference space of the element, which typically is spanned by a product of the interval ``[-1, 1]`` and in this particular example ``[-1, 1] \times [-1, 1]``. 
+In this space a local numbering of nodes and faces exists, i.e.
+
+
+![local element](./assets/local_element.svg)
+
+
+The example shows a local face ID ordering, defined as:
+
+```julia
+faces(::Lagrange{2,RefCube,1}) = ((1,2), (2,3), (3,4), (4,1))
+```
+
+Other face ID definitions [can be found in the src files](https://github.com/KristofferC/JuAFEM.jl/blob/8224282ab4d67cb523ef342e4a6ceb1716764ada/src/interpolations.jl#L154) in the corresponding `faces` dispatch.
+
+
+The highlighted face, i.e. the two lines from node ID 3 to 6 and from 6 to 9, on the right hand side of our test mesh can now be described as
+
+```julia
+julia> edges = [
+           (3,6),
+           (6,9)
+       ]
+
+julia> interpolation = Lagrange{2, RefTetrahedron, 1}()
```

Comment:
This looks a bit out of place. I suggest you move this to after the `compute_faceset` function in the next code block.

Reply:
yep, agree